### PR TITLE
v1.0: accepted os versions to run the Meilisearch binary

### LINKED ISSUE #2066: v1.0: Accepted OS versions to run the Meilisearch binary
- Define 
  - the different OS Meilisearch supports
  - minimal glibc version required to run the Meilisearch binary

## To do
Some may be missing
- [ ] Add a new section to the quick start or add to an existing section

## References
- Issue on core: https://github.com/meilisearch/meilisearch/issues/3109
- Discussion: https://github.com/meilisearch/product/discussions/547
- SME: @curquiza

### PR DESCRIPTION
**Pull request by maryamsulemani97**
closes #2066

### CODE REVIEW

File: learn/getting_started/supported_os.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Supported operating systems
+
+Please ensure your operating system is listed on this page before installing the Meilisearch binary. Refer to the [changelog](https://github.com/meilisearch/MeiliSearch/releases) for more information on operating system support changes.
+
+If the provided binaries don't work for you, we recommend compiling Meilisearch from the source files. If the compilation fails, Meilisearch is not compatible with your machine.
+
+::: warning
+Any operating system or version not listed on this page does not mean Meilisearch won't work with it at all, we just don't guarantee it.
```
Comment (curquiza) [inline learn/getting_started/supported_os.md:8]:
```suggestion
Any operating system or version not listed on this page does not mean Meilisearch won't work with it at all, we just don't guarantee it through our test suite.
```

File: learn/getting_started/quick_start.md [hunk 2]
```diff
@@ -10,6 +10,16 @@ Let's get started!
...
 We'll start with downloading and installing Meilisearch. You have the option to install Meilisearch locally or deploy it over a cloud service.
 
+### Supported operating systems
+
+The following operating systems are compatible with the Meilisearch binary:
+
+- Any Linux distribution with `amd64/x86_64` or `aarch64/arm64` architecture using glibc 2.27 and later
+- macOS 12 and later with `amd64` or `arm64` architecture
+- Windows 11 and later
+
```
[the remark this answers is not in the document]
Comment (curquiza) [reply learn/getting_started/quick_start.md:20]: Also docker is a good alternative 🐳

File: learn/getting_started/supported_os.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Supported operating systems
+
+Please ensure your operating system is listed on this page before installing the Meilisearch binary. Refer to the [changelog](https://github.com/meilisearch/MeiliSearch/releases) for more information on operating system support changes.
+
+If the provided binaries don't work for you, we recommend compiling Meilisearch from the source files. If the compilation fails, Meilisearch is not compatible with your machine.
```
[the remark this answers is not in the document]
Comment (curquiza) [reply learn/getting_started/supported_os.md:5]: And docker?

File: learn/getting_started/supported_os.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+::: warning
+Any operating system or version not listed on this page does not mean Meilisearch won't work with it at all, we just don't guarantee it.
+:::
+
+## Linux
+
+The Meilisearch binary works on all Linux distributions with `amd64/x86_64` or `aarch64/arm64` architecture using glibc 2.27 and later.
```
Comment (curquiza) [inline learn/getting_started/supported_os.md:13]: Maybe give the command to know the glibs version (I put in the engine team issue)?

File: learn/getting_started/supported_os.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Supported operating systems
+
+Please ensure your operating system is listed on this page before installing the Meilisearch binary. Refer to the [changelog](https://github.com/meilisearch/MeiliSearch/releases) for more information on operating system support changes.
```
Comment (curquiza) [inline learn/getting_started/supported_os.md:3]: Maybe give link about how to install meilisearch binaries?

**Review by curquiza — COMMENTED** [review]
Thanks @maryamsulemani97!

File: learn/getting_started/quick_start.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davelarkan) [reply learn/getting_started/quick_start.md:20]: I feel like there could be more mention of the Cloud in the quick start but it's probably something that requires more thought than just mentioning it in the `Supported operating systems` section.

For example, just above we say 
> We'll start with downloading and installing Meilisearch. You have the option to install Meilisearch locally or deploy it over a cloud service.

Which doesn't mention _our_ cloud service as an option.

Now that we have a free tier plan for testing Meilisearch we could consider Meilisearch Cloud as the fastest way to get started. And so perhaps the "Getting Started" area of the "Quick start" could have more of a cloud focus and much of the rest of this page could be moved to a separate "Installing and running Meilisearch" section.

That's quite a big change though and might not be something we feel is right strategically. I'm happy to hear what others think.

File: .vale/styles/Vocab/word_list/accept.txt [hunk 6]
```diff
@@ -78,3 +78,4 @@ AMIs
 [Cc]ancelation
 Brotli
 zlib
+glibc
```
[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply .vale/styles/Vocab/word_list/accept.txt:81]: This is used by [Vale](https://vale.sh/docs/vale-cli/overview/) (a prose linter). Vale checks for spelling as well. Some of the words we use in the docs don't exist in the dictionary. Adding them here means Vale doesn't show a warning for making a "typo".

File: learn/getting_started/quick_start.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply learn/getting_started/quick_start.md:20]: Hi! 👋 
We plan on restructuring the quick start after v1. The first step would be to have a [dedicated Installation page.](https://github.com/meilisearch/documentation/issues/2125) We will move this entire section along with the content in `supported_os.md` there. We do mention Docker in the Local installation section. 

Mentioning Meilisearch cloud is out of scope for this PR as it's dedicated to OS. But we will add it as part of restructuring the quick start.

File: learn/getting_started/supported_os.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (maryamsulemani97) [reply learn/getting_started/supported_os.md]: Same as [previous comment](https://github.com/meilisearch/documentation/pull/2112#discussion_r1081140455)

File: learn/getting_started/quick_start.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (curquiza) [reply learn/getting_started/quick_start.md:20]: Ok good to know! Thanks for the precision 😄

**Review by curquiza — APPROVED** [review]
Thanks @maryamsulemani97 😄

File: learn/getting_started/supported_os.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+If the provided [binaries](/learn/getting_started/quick_start.md#local-installation) don't work for you, we recommend compiling Meilisearch from the source files. If the compilation fails, Meilisearch is not compatible with your machine.
```
Comment (guimachiavelli) [inline learn/getting_started/supported_os.md:5]: Besides the minor edits, I would also recommend moving this paragraph so it's after the supported OS list. This way users get the list ASAP, and then only need to read what to do if something goes wrong/their OS is not listed.

```suggestion
If the provided [binaries](/learn/getting_started/quick_start.md#local-installation) don't work for you, try building Meilisearch from source. If compilation fails, Meilisearch is not compatible with your machine.
```

File: learn/getting_started/quick_start.md [hunk 8]
```diff
@@ -10,6 +10,16 @@ Let's get started!
 
 We'll start with downloading and installing Meilisearch. You have the option to install Meilisearch locally or deploy it over a cloud service.
 
+### Supported operating systems
+
+The following operating systems are compatible with the Meilisearch binary:
```
Comment (guimachiavelli) [inline learn/getting_started/quick_start.md:15]: Do users gain much from us specifying  compatibility with the binary? Can an OS be compatible with Meilisearch but incompatible with the binary?

```suggestion
The following operating systems are compatible with Meilisearch:
```

File: learn/getting_started/supported_os.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+::: warning
+Any operating system or version not listed on this page does not mean Meilisearch won't work with it at all, we just don't guarantee it through our test suite.
+:::
+
```
Comment (guimachiavelli) [inline learn/getting_started/supported_os.md:10]: I tried to integrate the info here on the first paragraph. The tone here is too informal for technical documentation, IMHO.

```suggestion
```

**Review by guimachiavelli — CHANGES_REQUESTED** [review]

File: learn/getting_started/supported_os.md [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+## macOS
+
+The Meilisearch binary works with macOS 12 and later with `amd64` or `arm64` architecture.
+
+## Windows
+
+The Meilisearch binary works with Windows 11 and later.
+
+If the provided [binaries](/learn/getting_started/quick_start.md#local-installation) don't work for you, try building Meilisearch from source. If compilation fails, Meilisearch is not compatible with your machine.
```
Comment (guimachiavelli) [inline learn/getting_started/supported_os.md:21]: Ok, I should have tested this on the preview before making my suggestion. Now it looks like it's part of the Windows instructions. I think the simplest solution is to return it to where it was, right below the first paragraph. Sorry 😅

**Review by guimachiavelli — CHANGES_REQUESTED** [review]
One final comment, rolling back on a previous suggestion. Should be good to go afterwards.

**Review by guimachiavelli — APPROVED** [review]
🆙

**Comment by guimachiavelli** [comment]
bors merge

**Pull request merged**